### PR TITLE
[ADD] purchase_requisition_type

### PR DESCRIPTION
**Pull request by mikevhe18**
Add order type to purchase requisition

### CODE REVIEW

**Comment by moylop260** [comment]
`purchase_requisition_type/__init__.py:1: [E7901(rst-syntax-error), ] README.rst:31 Duplicate implicit target name: "usage".`
https://travis-ci.org/OCA/purchase-workflow/jobs/130512494#L421

**Comment by mikevhe18** [comment]
@moylop260 Thank you for the review

**Comment by moylop260** [comment]
Thanks for contribution
👍

**Comment by andhit-r** [comment]
Error when encode a bid. Step to reproduce:
1. Create a requisition with **Regular** order type
2. Confirm requisition
3. Encode a bid.

Error message:

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/http.py", line 540, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/http.py", line 577, in dispatch
    result = self._call_function(*_self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/http.py", line 313, in _call_function
    return checked_call(self.db, *args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/service/model.py", line 118, in wrapper
    return f(dbname, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/http.py", line 310, in checked_call
    return self.endpoint(_a, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/http.py", line 806, in __call__
    return self.method(_args, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/http.py", line 406, in response_wrap
    response = f(_args, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/web/controllers/main.py", line 948, in call_button
    action = self._call_kw(model, method, args, {})
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 268, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 399, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase_requisition_bid_selection/wizard/purchase_requisition_partner.py", line 34, in create_order
    po_id = requisition.make_purchase_order(self.partner_id.id)[active_id]
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase_requisition_auto_rfq_bid_selection/model/purchase_requisition.py", line 33, in make_purchase_order
    return _super.make_purchase_order(seller_id)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 592, in new_api
    result = method(self._model, cr, uid, self.ids, _args, *_old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase_requisition/purchase_requisition.py", line 192, in make_purchase_order
    purchase_id = purchase_order.create(cr, uid, self._prepare_purchase_order(cr, uid, requisition, supplier, context=context), context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 268, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 372, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase_rfq_number/models/purchase.py", line 23, in create
    return super(PurchaseOrder, self).create(vals)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase_rfq_bid_workflow/model/purchase_order.py", line 149, in create
    order = super(PurchaseOrder, self).create(values)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase_order_revision/model/purchase_order.py", line 84, in create
    return super(PurchaseOrder, self).create(values)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 508, in new_api
    result = method(self._model, cr, uid, _args, *_old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/purchase/purchase.py", line 348, in create
    order =  super(purchase_order, self).create(cr, uid, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 268, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/mail/mail_thread.py", line 381, in create
    thread_id = super(mail_thread, self).create(cr, uid, values, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 268, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 372, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/addons/connector/producer.py", line 48, in create
    record_id = create_original(self, vals)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/models.py", line 4094, in create
    record = self.browse(self._create(old_vals))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 266, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/api.py", line 508, in new_api
    result = method(self._model, cr, uid, _args, *_old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/models.py", line 4232, in _create
    tuple([u[2] for u in updates if len(u) > 2])
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/sql_db.py", line 158, in wrapper
    return f(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3147261-231-bc3ff2/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
ProgrammingError: can't adapt type 'purchase.order.type'

**Comment by mikevhe18** [comment]
@andhit-r i've fixed the code. Thank You.

**Comment by andhit-r** [comment]
Hi @mikevhe18 . Thanks for the revision. The exception gone but another problem occur. Steps to reproduce:
1. Open **contract** purchase type. Change create invoice field into **Based on incoming shipment**
   selection_1
2. Create a call for bids with **contract** type
   selection_2
3. Encode a bid.
4. RFQ created has a correcy type (contract). But the invoice policy wrong, it should changed according purchase type's invoice policy.
   selection_3

**Comment by mikevhe18** [comment]
Hi @andhit-r, sorry i forgot to add invoice_method on _prepare_purchase_order function.
I've fixed the code.
Thanks in advance

File: purchase_requisition_type/tests/test_prepare_purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        self.assertIsNotNone(purchase_requisition)
+
+        # Check Purchase Term
+        self.assertIsNotNone(x)
+
+        # Check Prepare Purchase Order
+        data_prepare_po = self.obj_purchase_requisition.\
+            _prepare_purchase_order(purchase_requisition, self.supplier)
+
+        self.assertEqual(data_prepare_po.get('order_type'), self.order_type)
```
Comment (andhit-r) [inline purchase_requisition_type/tests/test_prepare_purchase_order.py]: @mikevhe18 can you assert invoice_method field also?

**Comment by andhit-r** [comment]
@mikevhe18 :+1:

**Comment by mikevhe18** [comment]
Hi @andhit-r i've already add assert inovice_method to the test_prepare_purchase_order.py

**Comment by azmimr67** [comment]
Has been tested
:+1:

**Comment by andhit-r** [comment]
Hi @OCA/procurement-purchase-maintainers , can we merge this PR?

**Comment by moylop260** [comment]
@andhit-r to avoid make builds red after merge... could fix the small flake error?
purchase_requisition_type/tests/test_prepare_purchase_order.py:58:80: E501 line too long (95 > 79 characters)

**Comment by mikevhe18** [comment]
Hi, @moylop260.
I'm already fix it.

**Comment by andhit-r** [comment]
Hi @moylop260 last commit from @mikevhe18 have cleared all lint error on travis. Would you kindly checked again? Thanks in advance.

**Comment by moylop260** [comment]
Thanks

**Comment by mikevhe18** [comment]
Thank you @moylop260

**Comment by moylop260** [comment]
Thank for contribute

**Pull request merged**